src/main/java/club/minnced/discord/rpc/DiscordRichPresence.java

```diff
@@ -178,6 +179,40 @@
      */
     public byte instance;
 
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        if (!super.equals(o))
+            return false;
+        DiscordRichPresence presence = (DiscordRichPresence) o;
+        return startTimestamp == presence.startTimestamp
+                && endTimestamp == presence.endTimestamp
+                && partySize == presence.partySize
+                && partyMax == presence.partyMax
+                && instance == presence.instance
+                && Objects.equals(state, presence.state)
+                && Objects.equals(details, presence.details)
+                && Objects.equals(largeImageKey, presence.largeImageKey)
+                && Objects.equals(largeImageText, presence.largeImageText)
+                && Objects.equals(smallImageKey, presence.smallImageKey)
+                && Objects.equals(smallImageText, presence.smallImageText)
+                && Objects.equals(partyId, presence.partyId)
+                && Objects.equals(matchSecret, presence.matchSecret)
+                && Objects.equals(joinSecret, presence.joinSecret)
+                && Objects.equals(spectateSecret, presence.spectateSecret);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(state, details, startTimestamp, endTimestamp, largeImageKey, largeImageText, smallImageKey,
```

Comment:
I believe the identity hashCode is better for JNA structures as they represent a pointer value

Reply:
That would destroy [the general contract of hashCode()](https://docs.oracle.com/javase/9/docs/api/java/lang/Object.html#hashCode()) though.
> If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result.

Reply:
I see, I still want the other structs to get the same treatment ^^